crates/ironrdp-graphics/src/dwt.rs

```diff
@@ -171,18 +190,28 @@ fn inverse_vertical(mut buffer: &mut [i16], mut temp_buffer: &[i16], subband_wid
             h = &h[total_width..];
 
             // Even coefficients
-            dst[2 * total_width] = (i32::from(l[0]) - ((i32::from(lh[0]) + i32::from(h[0]) + 1) >> 1)) as i16;
+            dst[2 * total_width] =
+                i32_to_i16_possible_truncation(i32::from(l[0]) - ((i32::from(lh[0]) + i32::from(h[0]) + 1) >> 1));
 
             // Odd coefficients
-            dst[total_width] =
-                (i32::from(lh[0] << 1) + ((i32::from(dst[0]) + i32::from(dst[2 * total_width])) >> 1)) as i16;
+            dst[total_width] = i32_to_i16_possible_truncation(
+                i32::from(lh[0] << 1) + ((i32::from(dst[0]) + i32::from(dst[2 * total_width])) >> 1),
+            );
 
             dst = &mut dst[2 * total_width..];
         }
 
-        dst[total_width] = (i32::from(lh[total_width] << 1) + ((i32::from(dst[0]) + i32::from(dst[0])) >> 1)) as i16;
+        dst[total_width] = i32_to_i16_possible_truncation(
+            i32::from(lh[total_width] << 1) + ((i32::from(dst[0]) + i32::from(dst[0])) >> 1),
+        );
 
         temp_buffer = &temp_buffer[1..];
         buffer = &mut buffer[1..];
     }
 }
+
+#[expect(clippy::as_conversions)]
+#[expect(clippy::cast_possible_truncation)]
+fn i32_to_i16_possible_truncation(value: i32) -> i16 {
```

Comment:
Added this helper usage, which performs `as` operations under the hood in this file, because value manipulations in this module are quite tricky, and I don't want to break anything. So go with a safe option here.